### PR TITLE
[Product Creation AI v2] Layout updates and Tone button

### LINKED ISSUE #13107: Tone selection button
- UI for tone selection button and popup menu.
- Logic can be reused from the existing tone selection button.

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #13107
Part of: #13106

## Description

Starting Information screen - Update layout, texts and add "Tone of voice" button.

Changes
- Update the generate product details button title.
- Update layout spacing to match new design.
- Add "Tone of voice" button. 
- Localize `AIToneVoice` description values. 

## Steps to reproduce

1. Create a JN site with Jetpack AI plugin
2. Turn on `productCreationAIv2M1` feature flag
3. Build and run the app
4. Navigate to the products tab
5. Tap `+` button on top right
6. Tap "Create a product with AI"
7.  Tap "Tone of voice" button below the text field
8. Validate the tone selection works and gets retained after app relaunch.
9. Validate that the layout matches the new design. 

## Screenshots

| Light | Dark |
|--------|--------|
| Simulator Screenshot - iPhone 15 Pro Max - 2024-07-02 at 21 38 35 | Simulator Screenshot - iPhone 15 Pro Max - 2024-07-02 at 21 39 00 | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/StartingInfo/ProductCreationAIStartingInfoView.swift [hunk 1]
```diff
@@ -275,13 +276,59 @@ private extension ProductCreationAIStartingInfoView {
     }
 }
 
+private extension ProductCreationAIStartingInfoView {
+    struct ToneOfVoiceView: View {
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/StartingInfo/ProductCreationAIStartingInfoView.swift]: ❓ Can this be on a separate file for clarity?

**Review by itsmeichigo — APPROVED** [review]
LGTM :shipit:

File: WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/StartingInfo/ProductCreationAIStartingInfoView.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/StartingInfo/ProductCreationAIStartingInfoView.swift]: Done in https://github.com/woocommerce/woocommerce-ios/pull/13225/commits/6c59cc29dedb9e30b0e33b4f4b78a9473b44a77d

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/212">19.4</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**